cmd/transflect-operator/metrics.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"k8s.io/client-go/util/workqueue"
+)
+
+var (
+	// filtersGauge includes filters that already exist at startup.
+	filtersGauge = promauto.NewGauge(prometheus.GaugeOpts{
+		Name: "transflect_envoyfilters",
+		Help: "Current number of EnvoyFilters managed by transflect.",
+	})
+	processedCounter = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "transflect_operations_total",
+		Help: "The total number of transflect operations processed.",
+	},
+		[]string{
+			"status", // success, error
+			"type",   // upsert, delete
+		},
+	)
+	preprocessErrCounter = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "transflect_preprocess_error_total",
+		Help: "The total number of replicaset ids popped off the queue that errored before being processed.",
+	})
+	ignoredCounter = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "transflect_ignored_total",
+		Help: "The total number of replicaset ids popped off the queue that were ignored and didn't need processing.",
+	})
+
+	// queue metrics.
+	qDepthGauge = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "transflect_queue_depth",
+		Help: "Current depth of the work queue.",
+	}, []string{"queue_name"})
+
+	qAddsCounter = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "transflect_queue_adds_total",
```

Comment:
I feel that these workqueue metrics should be named by the workqueue package so that they have consistent names across applications, but in the absense of that I wonder if you should name them with `workqueue_` as the prefix instead of `transflect_queue_`. i.e. they are not transflect metrics - their meaning is defined by the workqueue package.

For your other metrics, their meaning is defined by transflect so it makes sense to give them that prefix.

But this is a bit nitpicky - just pointing out a thought I had.

Reply:
No strong feelings about metric names, will update.